### ISSUE TITLE
Allow user to select a name for Dialogic files before creating them

### ISSUE THREAD

**Issue by lemonsaurus** [coppolaemilio/dialogic#256]
**Is your feature request related to a problem? Please describe.**
By default, new timelines, characters, themes and definitions are given a filename like `timeline_1285713812.json`. This makes code less readable when you want to load them, and generally adds clutter to my project.

**Describe the solution you'd like**
I would prefer to be able to select a unique name when I create these assets. A popup inputfield could ask me to specify a name for the Dialogic object, and this name could be used to create the filename - for example, if I name my object Orange Questionmark Room, the file could be named `orange_questionmark_room.json`. Conflicts could be resolved by appending numbers the way it is currently done.

**Describe alternatives you've considered**
I currently just create new characters and timelines by duplicating existing files and changing their names and contents, but this is tedious and shouldn't be necessary.

**Comment by coppolaemilio** [comment]
I agree that file management should be simplified. Not sure if it will come in the upcoming version but it will eventually be improved 👍🏻

**Comment by coppolaemilio** [comment]
Half of the issue was resolved in the latest commit: https://github.com/coppolaemilio/dialogic/commit/5ca0413383e7fa0c2098eb2c98cbbb94b9c767ad

Now, when creating new resources it will automatically start the renaming of it (similar to how it works on Visual Studio Code)
The other part to resolve would be the naming of the file in the filesystem.

**Comment by Jowan-Spooner** [comment]
I'm not sure how you want to "fix" that tho. I feel like it's a good decision that the file names are somewhat unspecific. I definitely think it should be made easier to find out what filename belongs to what character (I pointed it out here: #496), but I think if people would name their files themselves, it would result in big problems. Because imagine I name my timeline "Chapter 3" or my character "Laura". Then later when I might want to rename them, I would have to leave their filenames like this or break all existing links to the timeline/character.

Interested to hear your ideas though!

**Comment by coppolaemilio** [comment]
This can be fixed by adding the name that they chose + a random number/id generated like we do now. It will make it easier to know which file is which on the file system without compromising possible name collisions

**Comment by Jowan-Spooner** [comment]
This is outdated for 2.0, as timelines and characters are their own files now. Naming has to be done by the user.
